src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -57,90 +55,97 @@ const defaultProps = {
 
-    const submitForm = useCallback(() => {
-        const phoneLogin = LoginUtils.appendCountryCode(LoginUtils.getPhoneNumberWithoutSpecialChars(login));
-        const parsedPhoneNumber = parsePhoneNumber(phoneLogin);
-        const userLogin = parsedPhoneNumber.possible ? parsedPhoneNumber.number.e164 : login;
+        if (_.isEmpty(values.phoneOrEmail)) {
+            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', props.translate('contacts.genericFailureMessages.contactMethodRequired'));
+        }
 
-        // If this login already exists, just go back.
-        if (lodashGet(props.loginList, userLogin)) {
-            Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS);
-            return;
+        if (!_.isEmpty(values.phoneOrEmail) && !(parsePhoneNumber(phoneLogin).possible || Str.isValidEmail(values.phoneOrEmail))) {
+            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', props.translate('contacts.genericFailureMessages.invalidContactMethod'));
         }
 
-        User.addNewContactMethodAndNavigate(userLogin, password);
-    }, [login, props.loginList, password]);
+        if (!_.isEmpty(values.phoneOrEmail) && lodashGet(props.loginList, (values.phoneOrEmail).toLowerCase())) {
+            ErrorUtils.addErrorMessage(errors, 'phoneOrEmail', props.translate('contacts.genericFailureMessages.enteredMethodIsAlreadySubmited'));
+        }
+
+        if (!Permissions.canUsePasswordlessLogins(props.betas) && _.isEmpty(values.password)) {
+            errors.password = props.translate('contacts.genericFailureMessages.passwordRequired');
+        }
 
+        return errors;
+    };
+
+    const submitForm = (values) => {
+        const phoneLogin = getPhoneLogin(values.phoneOrEmail);
+        const parsedPhoneNumber = parsePhoneNumber(phoneLogin);
+        const userLogin = parsedPhoneNumber.possible ? parsedPhoneNumber.number.e164 : values.phoneOrEmail;
+
+        User.addNewContactMethodAndNavigate(userLogin.trim(), values.password);
+    };
     return (
         <ScreenWrapper
             onEntryTransitionEnd={() => {
                 if (!loginInputRef.current) {
                     return;
                 }
+
                 loginInputRef.current.focus();
             }}
+
+            includeSafeAreaPaddingBottom={false}
         >
             <HeaderWithCloseButton
                 title={props.translate('contacts.newContactMethod')}
                 shouldShowBackButton
                 onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS)}
                 onCloseButtonPress={() => Navigation.dismissModal(true)}
             />
-            <ScrollView>
-                <Text style={[styles.ph5, styles.mb5]}>
+            <Form
+                formID={ONYXKEYS.FORMS.NEW_CONTACT_METHOD_FORM}
+                validate={isFormValid}
+                onSubmit={submitForm}
+                submitButtonText={props.translate('common.add')}
+                style={[styles.flexGrow1, styles.mh5]}
```

Comment:
Since this form can be used in OFFLINE, we should add `enabledWhenOffline` to the Form and clear the draft in the optimistic data object.